packages/lib-classifier/src/components/Classifier/components/MetaTools/components/HidePreviousMarksButton/HidePreviousMarksButton.js

```diff
@@ -1,23 +1,38 @@
 import { MetaToolsButton } from '@zooniverse/react-components'
 import counterpart from 'counterpart'
 import PropTypes from 'prop-types'
-import { FormView, FormViewHide } from 'grommet-icons'
+import { FormView, FormViewHide, Hide } from 'grommet-icons'
 import React from 'react'
 
+import SHOWN_MARKS from '../../../../../../helpers/shownMarks'
 import en from './locales/en'
 
 counterpart.registerTranslations('en', en)
 
 export default function HidePreviousMarksButton (props) {
-  const { disabled, hidePreviousMarks, onClick } = props
-  const icon = hidePreviousMarks ? <FormViewHide /> : <FormView />
-  const text = hidePreviousMarks ?
-    counterpart('HidePreviousMarksButton.show') :
-    counterpart('HidePreviousMarksButton.hide')
+  const { disabled, onClick, shownMarks, type } = props
+  let text, icon;
+  const isDrawingTask = type === 'drawing'
+
+  switch(shownMarks) {
+    case SHOWN_MARKS.ALL:
+      text = isDrawingTask ? counterpart('HidePreviousMarksDrawingButton.hide')
+        : counterpart('HidePreviousMarksTranscriptionButton.showUser')
+      icon = <FormView />
+      break;
+    case SHOWN_MARKS.USER:
+      text = counterpart('HidePreviousMarksTranscriptionButton.hide')
+      icon = <Hide />
+      break;
+    default:
+      text = isDrawingTask ? counterpart('HidePreviousMarksDrawingButton.show')
+        : counterpart('HidePreviousMarksTranscriptionButton.show')
+      icon = <FormViewHide />
+  }
 
   return (
     <MetaToolsButton
-      aria-checked={hidePreviousMarks}
+      aria-checked={shownMarks}
```

Comment:
I think this is setting `checked` state to a number, but it can only be true/false.